src/network/clientpackethandler.cpp

```diff
@@ -958,114 +958,68 @@ void Client::handleCommand_SpawnParticle(NetworkPacket* pkt)
 	std::string datastring(pkt->getString(0), pkt->getSize());
 	std::istringstream is(datastring, std::ios_base::binary);
 
-	v3f pos                 = readV3F32(is);
-	v3f vel                 = readV3F32(is);
-	v3f acc                 = readV3F32(is);
-	float expirationtime    = readF32(is);
-	float size              = readF32(is);
-	bool collisiondetection = readU8(is);
-	std::string texture     = deSerializeLongString(is);
-
-	bool vertical          = false;
-	bool collision_removal = false;
-	TileAnimationParams animation;
-	animation.type         = TAT_NONE;
-	u8 glow                = 0;
-	bool object_collision  = false;
-	try {
-		vertical = readU8(is);
-		collision_removal = readU8(is);
-		animation.deSerialize(is, m_proto_ver);
-		glow = readU8(is);
-		object_collision = readU8(is);
-	} catch (...) {}
+	ParticleParameters p;
+
+	p.pos                = readV3F32(is);
```

Comment:
How about moving this to `ParticleParameters::deSerialize()`?

Reply:
Kinda wanted to avoid a second `particles.cpp`, but sure can do.

Reply:
Hm actually, for the particlespawner the code mixes between reading to `p` and reading stuff like `server_id` or `attached_id`. It's a little messy to pull that apart.
Should I do this for `ParticleParameters` only?

Reply:
Yes, I've seen the messy part, thus only suggested serialize functions for it for `ParticleParameters`. However, that could also be done at a later time.

Reply:
Really not sure if this is any better than before: https://github.com/minetest/minetest/pull/9893/commits/7806c2dd03e7999469740ac0f9afe0491baaeb26

Reply:
Thank you. It looks better to me because it properly separates the struct serialization from other variables.